lib/optimization.rb

```diff
@@ -1,12 +1,144 @@
 require "silicium"
+require 'fast_matrix'
 
 module Silicium
   module Optimization
 
+
     # reflector function
     def re_lu(x)
       x.negative? ? 0 : x
     end
-  end
 
+    #sigmoid function
+    def sigmoid(x)
+      1.0 / (1 + Math.exp(-x))
+    end
+
+    #integrating using method Monte Carlo (f - function, a, b - integrating limits, n - amount of random numbers)
+    def integrating_Monte_Carlo_base(a, b, n = 100000, &block)
+      res = 0
+      range = a..b.to_f
+      for i in 1..(n + 1)
+         x = rand(range)
+         res += (b - a) * 1.0 / n * block.call(x)
+      end
+      res
+    end
+
+    #return true if array is sorted
+    def sorted?(a)
+      return false if a.nil?
+      for i in 0..a.length - 2
+        if (a[i + 1] < a[i])
+          return false
+        end
+      end
+      true
+    end
+
+    #fastest(but it is not exactly) sort, modify sequance
+    def bogosort!(a)
+      if (a.nil?)
+        raise ArgumentError, "Nil array in bogosort"
```

Comment:
First, this would be better
 ```raise ArgumentError, "Nil array in bogosort"  if a.nil?```
Second, adding your own error class can make error handling more convenient.